src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java

```diff
@@ -0,0 +1,239 @@
+package io.confluent.maven.resolver;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Component;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+import org.codehaus.plexus.util.StringUtils;
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.RepositorySystemSession;
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.eclipse.aether.repository.RemoteRepository;
+import org.eclipse.aether.resolution.VersionRangeRequest;
+import org.eclipse.aether.resolution.VersionRangeResolutionException;
+import org.eclipse.aether.resolution.VersionRangeResult;
+import org.eclipse.aether.version.Version;
+
+/**
+ * This is a customized solution to resolves ce/ccs kafka version range to the
+ * highest matching version. By default, snapshots are excluded, as a workaround
+ * for <a href="https://issues.apache.org/jira/browse/MNG-3092">MNG-3092</a>.
+ *
+ * <p>
+ * For the version range format, see <a href=
+ * "https://github.com/eclipse/aether-core/blob/master/aether-util/src/main/java/org/eclipse/aether/util/version/GenericVersionScheme.java">
+ * GenericVersionScheme</a>
+ * </p>
+ *
+ * <p>
+ * Command-line example:
+ * </p>
+ *
+ * <pre>
+ * mvn io.confluent:resolver-maven-plugin:resolve-kafka-range \
+ *     -Dresolve.groupId=org.apache.kafka -Dresolve.artifactId=kafka-client "-Dresolve.versionRange=[6.0.0-1, 6.0.1-1]" \
+ *     -Dresolve.CEprint -q
+ * 	   -Dresolve.CCSprint -1
+ * </pre>
+ */
+@Mojo(name = "resolve-kafka-range", defaultPhase = LifecyclePhase.VALIDATE, threadSafe = true, requiresProject = false)
+public class ResolveKafkaVersionRangeMojo extends AbstractMojo {
+	private static final String SNAPSHOT = "SNAPSHOT";
+	private static final String CE = "ce-kafka";
+	private static final String CEVersionString = "-ce";
+	private static final String CCS = "ccs-kafka";
+	private static final String CCSVersionString = "-ccs";
+	private static final Pattern SNAPSHOT_TIMESTAMP = Pattern.compile("^(.*-)?([0-9]{8}\\.[0-9]{6}-[0-9]+)$");
+	private static final String CEKAFKAVERSION = "ce.kafka.version";
+	private static final String CCSKAFKAVERSION = "kafka.version";
+
+	/**
+	 * The group id of the artifact to resolve.
+	 */
+	@Parameter(property = "resolve.groupId", required = true)
```

Comment:
Can we drop the `resolve` prefix from all of these? It doesn't seem necessary? e.g. the other plugins we use don't seem to prefix stuff like this.